src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -62,16 +62,18 @@ def _post(self, request: Request):
             # Always compute the full config. It's invalid to send partial
             # configs to processing relays, and these validate the requests they
             # get with permissions and trim configs down accordingly.
-            response.update(self._post_or_schedule_by_key(request))
+            response.update(
+                # XXX(iker): we assume we can safely parse the version by
+                # casting it to an int if `_should_post_or_schedule` returns True
+                self._post_or_schedule_by_key(request, version=int(version))
```

Comment:
nit: I would do the conversion immediately on line 55.

Reply:
By not doing it on that line we guarantee the versions we get are actually correct, based on how Python handles casting:

```python
>>> int(3.5)
3
```

Casting after the validation function ensures the versions are expected (i.e. `3` is ok, `3.5` is not).